### PR TITLE
[SPARK-25897][k8s] Hook up k8s integration tests to sbt build.

### PR DESCRIPTION
**Pull request by vanzin**
The integration tests can now be run in sbt if the right profile
is enabled, using the "test" task under the respective project.

This avoids having to fall back to maven to run the tests, which
invalidates all your compiled stuff when you go back to sbt, making
development way slower than it should.

There's also a task to run the tests directly without refreshing
the docker images, which is helpful if you just made a change to
the submission code which should not affect the code in the images.

The sbt tasks currently are not very customizable; there's some
very minor things you can set in the sbt shell itself, but otherwise
it's hardcoded to run on minikube.

I also had to make some slight adjustments to the IT code itself,
mostly to remove assumptions about the existing harness.

Tested on sbt and maven.

### CODE REVIEW

File: resource-managers/kubernetes/integration-tests/pom.xml [hunk 1]
```diff
@@ -145,14 +145,10 @@
         <executions>
           <execution>
             <id>test</id>
+            <phase>none</phase>
             <goals>
               <goal>test</goal>
             </goals>
-            <configuration>
```
Comment (mccheah) [inline resource-managers/kubernetes/integration-tests/pom.xml]: Do we not need this anymore because the integration test is hidden behind the profile?

**Comment by mccheah** [comment]
Looks good to me but would like +1 from someone more familiar with SBT.

File: resource-managers/kubernetes/integration-tests/pom.xml (hunk 1 above)
Comment (vanzin) [reply resource-managers/kubernetes/integration-tests/pom.xml]: We don't need this because I disabled the execution the proper way a few lines above.

**Comment by mccheah** [comment]
Maybe @srowen on the SBT side?

**Comment by srowen** [comment]
I don't know SBT much myself. I would believe this works.

**Comment by vanzin** [comment]
sbt is kind of dark magic. read the right passages and things work. sometimes.

**Comment by vanzin** [comment]
BTW this is what it looks like (just to show it works):

```
> test
[ lots of output from compiling stuff]
[warn] /work/apache/spark/repl/src/main/scala/org/apache/spark/repl/SparkILoop.scala:54: method addedClasspath in class ILoop is deprecated: Use reset, replay or require to update class path
[warn]         settings.classpath append addedClasspath
[warn] 
Sending build context to Docker daemon  1.522GB
[lots of uninteresting output from docker]
Successfully built c643f12c5b88
Successfully tagged spark-r:dev
[info] Compiling 3 Scala sources to /work/apache/spark/resource-managers/kubernetes/integration-tests/target/scala-2.11/test-classes...
[success] created output: /work/apache/spark/resource-managers/kubernetes/integration-tests/target
[info] KubernetesSuite:
[info] - Run SparkPi with no resources (18 seconds, 58 milliseconds)
[info] - Run SparkPi with a very long application name. (17 seconds, 788 milliseconds)
[info] - Use SparkLauncher.NO_RESOURCE (17 seconds, 632 milliseconds)
[info] - Run SparkPi with a master URL without a scheme. (18 seconds, 288 milliseconds)
[info] - Run SparkPi with an argument. (19 seconds, 200 milliseconds)
[info] - Run SparkPi with custom labels, annotations, and environment variables. (17 seconds, 643 milliseconds)
[info] - Run extraJVMOptions check on driver (7 seconds, 481 milliseconds)
[info] - Run SparkRemoteFileTest using a remote data file (17 seconds, 809 milliseconds)
[info] - Run SparkPi with env and mount secrets. (29 seconds, 600 milliseconds)
[info] - Run PySpark on simple pi.py example (21 seconds, 703 milliseconds)
[info] - Run PySpark with Python2 to test a pyfiles example (19 seconds, 554 milliseconds)
[info] - Run PySpark with Python3 to test a pyfiles example (17 seconds, 629 milliseconds)
[info] - Run PySpark with memory customization (19 seconds, 817 milliseconds)
[info] - Run in client mode. (14 seconds, 537 milliseconds)
[info] - Start pod creation from template (17 seconds, 924 milliseconds)
[info] ScalaTest
[info] Total number of tests run: 15
[info] Suites: completed 1, aborted 0
[info] Tests: succeeded 15, failed 0, canceled 0, ignored 0, pending 0
[info] All tests passed.
[info] Passed: Total 15, Failed 0, Errors 0, Passed 15
[info] Run completed in 6 minutes, 30 seconds.
[success] Total time: 544 s, completed Nov 1, 2018 10:37:34 AM
```

I don't like how slow it is to build the images, but that's ok for now. At some point I might look into it more.

**Review by rvesse — APPROVED** [review]

**Comment by vanzin** [comment]
retest this please

**Comment by vanzin** [comment]
retest this please

**Comment by vanzin** [comment]
Ok, I'm merging this to master since there doesn't seem do be any negative feedback.

**Pull request closed without merging**